internal/zinc-core/src/main/scala/sbt/internal/inc/PickleJar.scala

```diff
@@ -20,7 +20,13 @@ import scala.reflect.io.RootPath
 
 object PickleJar {
   def write(pickleOut: Path, knownProducts: java.util.Set[String]): Unit = {
-    if (!Files.exists(pickleOut)) {
+    val pickleUpdatesJar = pickleOut.resolveSibling(s"${pickleOut.getFileName}-updates.jar")
+    if (Files.exists(pickleUpdatesJar)) {
+      if (Files.exists(pickleOut))
+        JarUtils.mergeJars(pickleOut.toFile, pickleUpdatesJar.toFile)
+      else
+        Files.move(pickleUpdatesJar, pickleOut)
+    } else if (!Files.exists(pickleOut)) {
```

Comment:
Would this fix the case of compiling `A`, then compiling `B`, and then do `use/compile` from another subproject?

Could you add repro case by adding something like https://github.com/sbt/zinc/tree/develop/zinc/src/sbt-test/source-dependencies/subproject-pipelining?

Reply:
Change dropped (superseded by the isPickleJava change)